sdk/core/core/src/az_url.c

```diff
@@ -7,12 +7,22 @@
 
 #include <_az_cfg.h>
 
-AZ_NODISCARD az_result
-az_url_emit_span_seq(az_url const * const self, az_span_action const write_span) {
-  AZ_CONTRACT_ARG_NOT_NULL(self);
-
-  AZ_RETURN_IF_FAILED(az_span_action_do(write_span, self->host));
-  AZ_RETURN_IF_FAILED(az_span_action_do(write_span, self->path));
-  AZ_RETURN_IF_FAILED(az_http_query_as_span_writer(self->query, write_span));
-  return AZ_OK;
+
+AZ_NODISCARD az_result az_url_parse(az_span const url, az_url* const out) {
+  AZ_CONTRACT_ARG_NOT_NULL(out);
+
+  (void)url;
+
+  return AZ_ERROR_NOT_IMPLEMENTED;
+}
+
+/**
+ * Read backwards (from right to left) from top-level domain (eg. `.com`) to the lowest subdomain
+ * (eg. `www`).
+ */
+AZ_NODISCARD az_result az_dns_read_domain(az_span* const dns, az_span* const subdomain) {
```

Comment:
It's not clear to me how a consumer calls this function; please show an example.

Reply:
I think the code is going to look like this - https://github.com/Azure/azure-sdk-for-c/pull/182/commits/aaf3f953f97d4dcbec346f796f0c692d5f37d3d1

(I am using it instead of the quick-and-dirty URL parser I wrote earlier today, and now you can see "before and after" code demonstration)